src/Model/FileExplorer.re

```diff
@@ -4,12 +4,14 @@ open Oni_Core;
 type t = {
   tree: option(FsTreeNode.t),
   isOpen: bool,
+  focus: option(string) // path
```

Comment:
The focus is stored as absolute filesystem path, since the node might not be loaded, or even in the workspace at all, and we probably want to preserve it if we change the workspace root too. It also makes it very easy to set in `BufferEnter` and elsewhere.